src/main/java/com/apptware/interview/immutability/Student.java

```diff
@@ -1,15 +1,46 @@
 /** This class is expected to be immutable. Please make necessary changes. */
 package com.apptware.interview.immutability;
 
+import java.util.ArrayList;
 import java.util.Date;
 import java.util.List;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 
-@Getter
-@AllArgsConstructor
+//@Getter
+//@AllArgsConstructor
 public class Student {
   private String name;
   private Date dateOfBirth;
   private List<String> courses;
+
+  public Student(String name, Date dateOfBirth, List<String> courses) {
+    this.name = name;
+    this.dateOfBirth = new Date(dateOfBirth.getTime());
```

Comment:
The issue in your test case is that you are directly modifying the mutable fields Date and List<String> after retrieving them from the Student object. To ensure immutability, you need to make sure that the student class provides defensive copies of these fields when accessed.